src/StreamsAPI_Lambda/Students/Main.java

```diff
@@ -0,0 +1,148 @@
+        students.add(new Students("Elon", "Gated", 27, "Male", "Mechanical Engineering", 90));
+        students.add(new Students("Justin", "Case", 26, "Male", "Computer Engineering", 340));
+        students.add(new Students("Dianne", "Ameter", 31, "Female", "Biotech Engineering", 128));
+        students.add(new Students("Joss", "Sticks", 23, "Male", "Computer Engineering", 20));
+
+        List <Students> filteredStudents = students.stream()
+                .filter(a -> a.getFirstName().startsWith("A"))//Find a list of students whose first name starts with 'A'
+                .collect(toList());
+        System.out.println("\n Name whose first name starts with A");
+        for (Students s : filteredStudents) {
+            System.out.println(s);
+        }
+
+        Map<String, Set<String>> namesByDepertament   = students.stream()
+                .collect(groupingBy(
+                        Students::getDepartamentName,
+                        mapping(
+                                Students::getFirstName,
+                                toSet())
+                ));
+        System.out.println("\n Group the students by department names");
+        namesByDepertament.forEach((departamentName,lastName) -> System.out.println(departamentName + " : " + lastName));
+
+        //total students
+        long total = students.stream().count();
+        System.out.println("\n Total number of students: " + total);
+
+        //maAge
+        Comparator<Students> studentsComparator = Comparator.comparing(Students::getAge);
+        Students maxAge = students.stream().max(studentsComparator).get();
+        System.out.println("Max age: " + maxAge);
+
+        //departaments names
+        Map<String, Set<String>> namessByDepertament   = students.stream()
+                .collect(groupingBy(
+                        Students::getDepartamentName,
+                        mapping(
+                                Students::getDepartamentName,
+                                toSet())
+                ));
+        System.out.println("\n Group  by department names");
+        namesByDepertament.forEach((departamentName, d) -> System.out.println(departamentName));
+
+
+        // count of students in each department
+        Map<String, Long> studentsByDepertament   = students.stream()
+                .collect(groupingBy(
+                        Students::getDepartamentName,//grupam dupa numele departamentuli
+                        counting()) //numaram membrii
+                );
+        System.out.println("\n Count department ");
+        studentsByDepertament.forEach((departament,count) -> System.out.println(departament + " : " +count));
+
+        //Find the list of students whose age is less than 30
+
+        List <Students> studentsUnder30 = students.stream()
+                .filter(students1 -> students1.getAge() < 30)//Find a list of students whose first name starts with 'A'
```

Comment:
note that `students1` is actually one student in your list. so you might want to name it `s` or `student`